packages/notifications/push/amplify_push_notifications/ios/Classes/AmplifyPushNotificationsPlugin.swift

```diff
@@ -1,19 +1,114 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
 import Flutter
-import UIKit
-
-public class AmplifyPushNotificationsPlugin: NSObject, FlutterPlugin {
-  public static func register(with registrar: FlutterPluginRegistrar) {
-    let channel = FlutterMethodChannel(name: "amplify_push_notifications", binaryMessenger: registrar.messenger())
-    let instance = AmplifyPushNotificationsPlugin()
-    registrar.addMethodCallDelegate(instance, channel: channel)
-  }
-
-  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
-    switch call.method {
-    case "getPlatformVersion":
-      result("iOS " + UIDevice.current.systemVersion)
-    default:
-      result(FlutterMethodNotImplemented)
-    }
-  }
+import amplify_flutter_ios
+
+public class AmplifyPushNotificationsPlugin: NSObject, FlutterPlugin, NativePushNotificationsPlugin {
+    var remoteNotificationCompletionHandlers: [String: (UIBackgroundFetchResult) -> Void] = [:]
+    // Set to true if the App was awaken in the background by a remote notification from the terminated state
+    var awakenInBackground = false
+
+    // The notification that an end user has tapped on and luanch the App from
+    //   1. terminated state
+    //   2. has been awaken in the background from the terminated state
+    // Can be null and set only when 1 or 2 is satisfied
+    var launchNotification: [AnyHashable: Any]?
+
+    let sharedEventsStreamHandlers: EventsStreamHandlers
+
+    private init(sharedEventsStreamHandlers: EventsStreamHandlers) {
+        self.sharedEventsStreamHandlers = sharedEventsStreamHandlers
+    }
+
+    public static func register(with registrar: FlutterPluginRegistrar) {
+        let messenger = registrar.messenger()
+
+        EventsStreamHandlers.binaryMessenger = messenger
+        let pluginInstance = AmplifyPushNotificationsPlugin(
+            sharedEventsStreamHandlers: EventsStreamHandlers.shared
+        )
+
+        NativePushNotificationsPluginSetup(messenger, pluginInstance)
+
+        registrar.addApplicationDelegate(pluginInstance)
+    }
+
+    public func getPermissionStatus(completion: @escaping (String?, FlutterError?) -> Void) {
+        UNUserNotificationCenter.current().getNotificationSettings { settings in
+            var status: String
```

Comment:
Pigeon supports enums - would prefer that over strings

Reply:
Yea, was thinking Android and iOS returns different values implementing this API, could aggregate and remap the result on Dart side, could make the enum a super set of all the value that may return on Android and iOS, but still need to remap somehow.

Reply:
Updated 👍